rules/prefer-nullish-coalescing.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+const {getStaticValue} = require('eslint-utils');
+const {matches} = require('./selectors/index.js');
+const {isBooleanNode} = require('./utils/boolean.js');
+
+const ERROR = 'error';
+const SUGGESTION = 'suggestion';
+const messages = {
+	[ERROR]: 'Prefer `{{replacement}}` over `{{original}}`.',
+	[SUGGESTION]: 'Use `{{replacement}}`'
+};
+
+const selector = matches([
+	'LogicalExpression[operator="||"]',
+	'AssignmentExpression[operator="||="]'
+]);
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	return {
+		[selector](node) {
+			if (
+				[node.parent, node.left, node.right].some(node => node.type === 'LogicalExpression') ||
+				isBooleanNode(node) ||
+				isBooleanNode(node.left)
+			) {
+				return;
+			}
+
+			const {left} = node;
+
+			const staticValue = getStaticValue(left, context.getScope());
+			if (staticValue) {
+				const {value} = staticValue;
+				if (!(typeof value === 'undefined' || value === null)) {
+					return;
+				}
+			}
+
+			const isAssignment = node.type === 'AssignmentExpression';
+			const originalOperator = isAssignment ? '||=' : '||';
+			const replacementOperator = isAssignment ? '??=' : '??';
+			const operatorToken = context.getSourceCode()
+				.getTokenAfter(
+					node.left,
+					token => token.type === 'Punctuator' && token.value === originalOperator
+				);
+
+			const messageData = {
+				original: originalOperator,
+				replacement: replacementOperator
+			};
+			return {
+				node: operatorToken,
+				messageId: ERROR,
+				data: messageData,
+				suggest: [
+					{
+						messageId: SUGGESTION,
+						data: messageData,
+						fix: fixer => fixer.replaceText(operatorToken, replacementOperator)
+					}
+				]
+			};
+		}
+	};
+};
+
+module.exports = {
```

Comment:
We'll want to add a `type` annotation here (more info in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1606):

```suggestion
/** @type {import('eslint').Rule.RuleModule} */
module.exports = {
```